diff --git a/build/modern/hydration.js b/build/modern/hydration.js
index 9e6d2397213ab404e34804315ecb167a16d0a2fc..377a63847f9a5412c29a05d22409ffc66263eac3 100644
--- a/build/modern/hydration.js
+++ b/build/modern/hydration.js
@@ -85,7 +85,14 @@ function hydrate(client, dehydratedState, options) {
       );
     }
     if (promise) {
-      void query.fetch(void 0, { initialPromise: promise });
+      const { deserialize = (x) => x } = client.getDefaultOptions().hydrate ?? {};
+
+      const initialPromise = (async () => {
+        const queryResult = await promise;
+        return deserialize(queryResult);
+
+      })()
+      void query.fetch(void 0, { initialPromise });
     }
   });
 }
