diff --git a/build/legacy/infiniteQueryOptions.d.cts b/build/legacy/infiniteQueryOptions.d.cts
index f8ff23086e0c7f5f3c9bec6acd986b984d5acd05..290a2b0f250f6260ae43e7098e9e290baa3ff764 100644
--- a/build/legacy/infiniteQueryOptions.d.cts
+++ b/build/legacy/infiniteQueryOptions.d.cts
@@ -6,7 +6,10 @@ type UndefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TD
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown> = UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryFnData, TQueryKey, TPageParam> & {
-    initialData: NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>> | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>);
+    initialData:
+    | NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
+    | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)
+    | undefined
 };
 declare function infiniteQueryOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown>(options: DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam>): DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam> & {
     queryKey: DataTag<TQueryKey, InfiniteData<TQueryFnData>>;
diff --git a/build/legacy/infiniteQueryOptions.d.ts b/build/legacy/infiniteQueryOptions.d.ts
index 89f7f25698efa5cfb6336e31f0ef7ed0f4a8a6bb..2e7a7bc37887399b7c4693ca50d65990b35e6e79 100644
--- a/build/legacy/infiniteQueryOptions.d.ts
+++ b/build/legacy/infiniteQueryOptions.d.ts
@@ -6,7 +6,10 @@ type UndefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TD
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown> = UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryFnData, TQueryKey, TPageParam> & {
-    initialData: NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>> | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>);
+    initialData:
+    | NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
+    | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)
+    | undefined
 };
 declare function infiniteQueryOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown>(options: DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam>): DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam> & {
     queryKey: DataTag<TQueryKey, InfiniteData<TQueryFnData>>;
diff --git a/build/legacy/queryOptions.d.cts b/build/legacy/queryOptions.d.cts
index d5ae2aa79586593fe0cc92a09a461d022f1702ea..e540084f9e6796dad9604b8d0c3d72d16420d094 100644
--- a/build/legacy/queryOptions.d.cts
+++ b/build/legacy/queryOptions.d.cts
@@ -2,7 +2,10 @@ import { DefaultError, QueryKey, InitialDataFunction, DataTag } from '@tanstack/
 import { UseQueryOptions } from './types.cjs';
 
 type UndefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
-    initialData?: undefined | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>;
+    initialData?:
+    | undefined
+    | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>
+    | NonUndefinedGuard<TQueryFnData>
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
diff --git a/build/legacy/queryOptions.d.ts b/build/legacy/queryOptions.d.ts
index 2316717958decc34095afb7ca0b1de7a79f52922..f2811d72fef9704ab43db95cdb7519c5abce404d 100644
--- a/build/legacy/queryOptions.d.ts
+++ b/build/legacy/queryOptions.d.ts
@@ -2,7 +2,10 @@ import { DefaultError, QueryKey, InitialDataFunction, DataTag } from '@tanstack/
 import { UseQueryOptions } from './types.js';
 
 type UndefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
-    initialData?: undefined | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>;
+    initialData?:
+    | undefined
+    | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>
+    | NonUndefinedGuard<TQueryFnData>
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
diff --git a/build/modern/infiniteQueryOptions.d.cts b/build/modern/infiniteQueryOptions.d.cts
index f8ff23086e0c7f5f3c9bec6acd986b984d5acd05..290a2b0f250f6260ae43e7098e9e290baa3ff764 100644
--- a/build/modern/infiniteQueryOptions.d.cts
+++ b/build/modern/infiniteQueryOptions.d.cts
@@ -6,7 +6,10 @@ type UndefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TD
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown> = UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryFnData, TQueryKey, TPageParam> & {
-    initialData: NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>> | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>);
+    initialData:
+    | NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
+    | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)
+    | undefined
 };
 declare function infiniteQueryOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown>(options: DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam>): DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam> & {
     queryKey: DataTag<TQueryKey, InfiniteData<TQueryFnData>>;
diff --git a/build/modern/infiniteQueryOptions.d.ts b/build/modern/infiniteQueryOptions.d.ts
index 89f7f25698efa5cfb6336e31f0ef7ed0f4a8a6bb..2e7a7bc37887399b7c4693ca50d65990b35e6e79 100644
--- a/build/modern/infiniteQueryOptions.d.ts
+++ b/build/modern/infiniteQueryOptions.d.ts
@@ -6,7 +6,10 @@ type UndefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TD
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataInfiniteOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown> = UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryFnData, TQueryKey, TPageParam> & {
-    initialData: NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>> | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>);
+    initialData:
+    | NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
+    | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)
+    | undefined
 };
 declare function infiniteQueryOptions<TQueryFnData, TError = DefaultError, TData = InfiniteData<TQueryFnData>, TQueryKey extends QueryKey = QueryKey, TPageParam = unknown>(options: DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam>): DefinedInitialDataInfiniteOptions<TQueryFnData, TError, TData, TQueryKey, TPageParam> & {
     queryKey: DataTag<TQueryKey, InfiniteData<TQueryFnData>>;
diff --git a/build/modern/queryOptions.d.cts b/build/modern/queryOptions.d.cts
index d5ae2aa79586593fe0cc92a09a461d022f1702ea..e540084f9e6796dad9604b8d0c3d72d16420d094 100644
--- a/build/modern/queryOptions.d.cts
+++ b/build/modern/queryOptions.d.cts
@@ -2,7 +2,10 @@ import { DefaultError, QueryKey, InitialDataFunction, DataTag } from '@tanstack/
 import { UseQueryOptions } from './types.cjs';
 
 type UndefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
-    initialData?: undefined | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>;
+    initialData?:
+    | undefined
+    | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>
+    | NonUndefinedGuard<TQueryFnData>
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
diff --git a/build/modern/queryOptions.d.ts b/build/modern/queryOptions.d.ts
index 2316717958decc34095afb7ca0b1de7a79f52922..f2811d72fef9704ab43db95cdb7519c5abce404d 100644
--- a/build/modern/queryOptions.d.ts
+++ b/build/modern/queryOptions.d.ts
@@ -2,7 +2,10 @@ import { DefaultError, QueryKey, InitialDataFunction, DataTag } from '@tanstack/
 import { UseQueryOptions } from './types.js';
 
 type UndefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
-    initialData?: undefined | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>;
+    initialData?:
+    | undefined
+    | InitialDataFunction<NonUndefinedGuard<TQueryFnData>>
+    | NonUndefinedGuard<TQueryFnData>
 };
 type NonUndefinedGuard<T> = T extends undefined ? never : T;
 type DefinedInitialDataOptions<TQueryFnData = unknown, TError = DefaultError, TData = TQueryFnData, TQueryKey extends QueryKey = QueryKey> = UseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
