diff --git a/build/modern/hydration.js b/build/modern/hydration.js
index c8400cc568d659145206d4e6672529566f0ae5b3..88a9226b2c3048fa3f7e8c4b8b300ef6f0b7b3f7 100644
--- a/build/modern/hydration.js
+++ b/build/modern/hydration.js
@@ -67,6 +67,8 @@ function hydrate(client, dehydratedState, options) {
         query.setState(dehydratedQueryState);
       }
     } else {
+      console.log("QUERY", query, state)
+      const transformData = client.getDefaultOptions().hydrate?.transformData;
       query = queryCache.build(
         client,
         {
@@ -80,9 +82,11 @@ function hydrate(client, dehydratedState, options) {
         // query being stuck in fetching state upon hydration
         {
           ...state,
+          ...(state.data && { data: transformData?.(state.data) ?? state.data }),
           fetchStatus: "idle"
         }
       );
+      console.log("QUERY", query)
     }
     if (promise) {
       const transformPromise = client.getDefaultOptions().hydrate?.transformPromise;
